envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -62,23 +71,39 @@ internal class EnvoySnapshotFactory(
         return newSnapshotForGroup(group, globalSnapshot)
     }
 
-    private fun getServiceNamesForGroup(group: Group, globalSnapshot: Snapshot): List<String> {
+    private fun getRouteSpecificationsForGroup(group: Group, globalSnapshot: Snapshot): Collection<RouteSpecification> {
         return when (group) {
-            is ServicesGroup -> group.proxySettings.outgoing.getServiceDependencies().map { it.service }
-            is AllServicesGroup -> globalSnapshot.clusters().resources().map { it.key }
+            is ServicesGroup -> group.proxySettings.outgoing.getServiceDependencies().map {
+                RouteSpecification(
+                    clusterName = it.service,
+                    routeDomain = it.service,
+                    settings = it.settings
+                )
+            }
+            is AllServicesGroup -> globalSnapshot.clusters().resources().map {
+                RouteSpecification(
+                    clusterName = it.key,
+                    routeDomain = it.key,
+                    settings = defaultDependencySettings
+                )
+            }
         }
     }
 
-    private fun getEgressRouteMap(group: Group, globalSnapshot: Snapshot): Map<String, String> {
-        return getServiceNamesForGroup(group, globalSnapshot).map { it to it }.toMap() +
+    private fun getEgressRoutesSpecification(group: Group, globalSnapshot: Snapshot): Collection<RouteSpecification> {
+        return getRouteSpecificationsForGroup(group, globalSnapshot) +
             group.proxySettings.outgoing.getDomainDependencies().map {
```

Comment:
Now this inline transformation deserves its' own method: `getDomainRouteSpecifications` and will play along nicely with the one for services suggested above.